driver-core/src/main/com/mongodb/TaggableReadPreference.java

```diff
@@ -286,7 +312,34 @@ private boolean serversAreAllThreeDotFour(final ClusterDescription clusterDescri
         }
 
         SecondaryReadPreference(final List<TagSet> tagSetList, @Nullable final Long maxStaleness, final TimeUnit timeUnit) {
-            super(tagSetList, maxStaleness, timeUnit);
+            this(tagSetList, maxStaleness, timeUnit, null);
+        }
+
+        SecondaryReadPreference(final List<TagSet> tagSetList, @Nullable final Long maxStaleness, final TimeUnit timeUnit,
+                                @Nullable final ReadPreferenceHedgeOptions hedgeOptions) {
+            super(tagSetList, maxStaleness, timeUnit, hedgeOptions);
+        }
+
+        @Override
+        public SecondaryReadPreference withTagSet(final TagSet tagSet) {
```

Comment:
We definitely should not make the return type `SecondaryReadPreference`, because that class is not public.  I think it's safer to make it `TaggableReadPreference`, as in the methods being overridden.

I just asked in `#driver-devs` about a better name than `TaggableReadPreference`, but so far nothing better has come up.  The idea is to introduce a new superclass between `TaggableReadPreference` and `ReadPreference` that describes better what it is.  When we first came up with the `TaggableReadPreference` class name, tag set lists were the only "extra" feature of non-primary read preferences.  But since then we've added max staleness, and now hedging, so the name no longer makes as much sense.  If we could come up with something, that would be the appropriate type to return from these methods.  If we can't, then `TaggableReadPreference` is probably best.

Reply:
Since there was no consensus on the `#driver-devs` channel, I have kept the name `TaggableReadPreference`, and I have changed the return types for the `with...` methods accordingly.

Reply:
Sounds good